### ISSUE TITLE
Performance hit with large amount of .md-ellipsis elements

### ISSUE THREAD

**Issue by quarckster** [squidfunk/mkdocs-material#7317]
### Bug description

I have a big documentation site with a section that has more than 5000 navigation items. Page loading takes a significant amount of time due to [patchEllipsis](https://github.com/squidfunk/mkdocs-material/blob/12a8e82837fe400dd1f123e41d75b32987a11744/src/templates/assets/javascripts/patches/ellipsis/index.ts#L68C17-L68C30) is applied to all `.md-ellipsis` elements. I attached profiler reports from Chrome and Firefox.

[Trace-20240701T115430.json](https://github.com/user-attachments/files/16051107/Trace-20240701T115430.json)
[Firefox 2024-07-01 11.52 profile.json.gz](https://github.com/user-attachments/files/16051109/Firefox.2024-07-01.11.52.profile.json.gz)

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Navigation pruning](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#navigation-pruning)

### Reproduction

[9.5.27-md-ellipsis-performance-hit.zip](https://github.com/user-attachments/files/16051292/9.5.27-md-ellipsis-performance-hit.zip)

### Steps to reproduce

1. Create a section in navigation with 5000+ items.
2. Run `mkdocs serve`.
3. Open a page in the section.

### Browser

Chrome, Firefox

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting and providing the reproduction, we'll look into it. However, I'm not sure whether this is fixable. 5,000 is quite a number. It might be a good idea to nest navigation items to make use of navigation pruning or use navigation tabs to add some more structure.

**Comment by quarckster** [comment]
I believe the workaround or maybe fix is quite simple, just disable `patchEllipsis` patch and add `title` attribute to all links in [nav-item](https://github.com/squidfunk/mkdocs-material/blob/12a8e82837fe400dd1f123e41d75b32987a11744/src/templates/partials/nav-item.html#L276). Browser tooltip is more than enough.

**Comment by squidfunk** [comment]
I've checked your reproduction and I cannot see a sidebar or navigation where the problem is visible. It does not seem to be a minimal reproduction, because it does not show the problem described. Furthermore, the solution you proposed does not provide for a good user experience. If we would add a title attribute to all links, every link would show the same content as the link text itself, which is irritating:

<img width="142" alt="Bildschirm­foto 2024-07-01 um 14 29 01" src="https://github.com/squidfunk/mkdocs-material/assets/932156/e3471be8-655e-4303-8812-5fd24f402803">

It only makes sense to add tooltips when the text is cut by an ellipsis. I realized that we don't have a check for the [`content.tooltips`](https://squidfunk.github.io/mkdocs-material/reference/tooltips/#improved-tooltips) feature in place, which I just added in 845fc610b, so browser tooltips will be used if you do not enable this feature. This might also improve the situation.

Thus, we can consider this problem resolved. Also, I strongly recommend restructuring your site. 5,000 navigation items on the same level is just too much to be useful and definitely an edge case we are currently not optimizing for 😉
